src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -244,7 +244,8 @@ const HierarchyList = ({
     cancelMoveClicked();
   };
 
-  useEffect(
+  useDeepCompareEffect(
```

Comment:
We're already running a deep compare effect anytime the items change up on line 209. Would be best to encapsulate this logic there together so we're not deep comparing the items twice.

Reply:
Also worth noting that this opens the HierarchyList up to a potential out of memory error if `items` ever contains a React.Element or Node. useDeepCompareEffect can't compare props of these types.

Reply:
hmm unfortunately I can't because I need a different trigger for the second logic (i.e. I only want it to trigger if the selectedId changes). @tay1orjones.  The first one should trigger every time the items change

Reply:
React [will bail out](https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update) of the upper `setFilteredItems(items);` state call if `items` is the same, so it should be fine to run `setFilteredItems(items);` when either `defaultSelectedId` or `items` changes.

Reply:
Ah nevermind, I see what you're saying. This is just really expensive and is very likely to cause perf issues with heavily nested hierarchies. Is there any option to prevent the infinite loop upstream before passing new `items` to the component?

Reply:
it is possible to memoize the items so the same object is passed each time, but I think the symptom is soo bad unfortunately it's probably better to protect it for external consumers of the component @tay1orjones

Reply:
Yeah, it's an unfortunate consequence of having these monolith prop objects. Let's merge this as-is to unblock you. I do want to follow up with you to see more specifically what's causing the infinite loop in your project.

I've opened https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1797 to look at this in regards to the comparison of nodes. As soon as you upgrade your babel deps to >=7.12.x, you'll get an out of memory error.